src/m3ninx/index/segment/builder/builder_test.go

```diff
@@ -105,6 +106,7 @@ func TestBuilderFields(t *testing.T) {
 
 func TestBuilderTerms(t *testing.T) {
 	builder, err := NewBuilderFromDocuments(testOptions)
+	defer builder.Close()
```

Comment:
Since this is a test, instead of deferring close here, maybe add `require.NoError(t, builder.Close())` at the end of the test?